File: api/src/main/java/net/blueshell/api/mapper/survey/AnswerMapper.java
package net.blueshell.api.mapper.survey;

import lombok.extern.slf4j.Slf4j;
import net.blueshell.api.base.BaseMapper;
import net.blueshell.api.dto.survey.AnswerDTO;
import net.blueshell.api.model.survey.Answer;
import org.mapstruct.*;

@Slf4j
@Mapper(componentModel = "spring")
public abstract class AnswerMapper extends BaseMapper<Answer, AnswerDTO> {
    @Mapping(target = "id")
    @Mapping(target = "questionId")
    @Mapping(target = "optionSelections")
    @Mapping(target = "textResponse")
    @BeanMapping(ignoreByDefault = true)
    public abstract Answer fromDTO(AnswerDTO dto, @MappingTarget Answer answer);

    @Mapping(target = "id")
    @Mapping(target = "questionId")
    @Mapping(target = "optionSelections")
    @Mapping(target = "textResponse")
    @BeanMapping(ignoreByDefault = true)
    public abstract AnswerDTO toDTO(Answer answer);
}

File: api/src/main/kotlin/net/blueshell/api/mapper/survey/AnswerMapper.kt
package net.blueshell.api.mapper.survey

import net.blueshell.api.base.BaseMapper
import net.blueshell.api.dto.survey.AnswerDTO
import net.blueshell.api.model.survey.Answer
import org.mapstruct.BeanMapping
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingTarget

@Mapper(componentModel = "spring")
abstract class AnswerMapper : BaseMapper<Answer?, AnswerDTO?>() {
    @Mapping(target = "id")
    @Mapping(target = "questionId")
    @Mapping(target = "optionSelections")
    @Mapping(target = "textResponse")
    @Mapping(target = "version")
    @BeanMapping(ignoreByDefault = true)
    abstract fun fromDTO(dto: AnswerDTO?, @MappingTarget answer: Answer?): Answer?

    @Mapping(target = "id")
    @Mapping(target = "questionId")
    @Mapping(target = "optionSelections")
    @Mapping(target = "textResponse")
    @Mapping(target = "version")
    @BeanMapping(ignoreByDefault = true)
    abstract override fun toDTO(answer: Answer?): AnswerDTO?
}
